fix: prevent 404.html override when routes exist

avernikoz opened the pull request:
When a website has a custom 404.html (e.g., from Next.js build), we incorrectly fetch it without checking routes first. This breaks routing functionality for websites that include 404.html in their build output. Modified route resolution order and added test case to verify proper behavior.

Fixes #400

Tzal3x commented on portal/common/lib/routing.test.ts:
```diff
@@ -51,3 +53,46 @@ testCases.forEach(([requestPath, _]) => {
+        const expectedContent = await directResponse.text();
+
+        // Now test the routing flow
+        const routedResponse = await urlFetcher.resolveDomainAndFetchUrl({
+            subdomain: siteObjectId,
+            path: "/test"
+        }, siteObjectId);
+        expect(routedResponse.status).toBe(200);
+        const actualContent = await routedResponse.text();
+
+        // Verify we got the same content as direct fetch
+        expect(actualContent).toBe(expectedContent);
+
+        // Also fetch 404.html to prove we got different content
+        const notFoundResponse = await urlFetcher.resolveDomainAndFetchUrl({
+            subdomain: siteObjectId,
+            path: "/404.html"
+        }, siteObjectId);
```
Please mock the result of `urlFetcher.fetchUrl` that gets called inside `resolveDomainAndFetchUrl`. 
We would like to keep the tests network request-independent.

If you need an example on how to do this, check [resource.test.ts](https://github.com/MystenLabs/walrus-sites/blob/main/portal/common/lib/resource.test.ts).

avernikoz replied:
Hey @Tzal3x ,  I've mocked `urlFetcher.fetchUrl` as well as the `wsRouter.getRoutes` to prevent network requests in tests.

Please, let me know if that's looking good for you.

Tzal3x reviewed (changes requested):
Thank you for the fix! Two things to add:

1.  Please rebase your branch to our `testnet` branch. There have been some changes in the meantime like including an aggregator Url as an attribute to the UrlFetcher etc. 
2. We need tests to be independent from network requests, so wherever possible, the responses should be mocked.

avernikoz commented:
Thanks for the review, going to look into the comments asap.

avernikoz commented:
> Thank you for the fix! Two things to add:
> 
> 1. Please rebase your branch to our `testnet` branch. There have been some changes in the meantime like including an aggregator Url as an attribute to the UrlFetcher etc.
> 2. We need tests to be independent from network requests, so wherever possible, the responses should be mocked.
> 
> Other than that, nice work the code looks clean!

Hey, @Tzal3x , thanks for the feedback! I've rebased onto testnet and updated the tests to be network-independent. 
Let me know if this looks good!

Tzal3x reviewed (approved):
Great! Thank you for your contribution @avernikoz ! 💯
